Disable importTags so user can be registered

hermamat opened the pull request:
Comment out part where KOSApi is called, because KOSapi doesn't support LDAP anymore and need to be called through OAuth2.0.
Retrieving current subjects through KOSapi is not crucial so this action is disabled, until new connection through OAuth2.0 will be programmed.

hermamat committed:
Switch method order in AuthPresenter

hermamat committed:
Fix typo in signUpAfter.latte

hermamat committed:
Disable (comment) importTags action when registering

KOSApi removed LDAP access, now it's OAuth 2.0 only.
It will take some time to make it work through OAuth.
Now KOSApi is needed only for retrieving current subjects for registered student,
 which is not crucial, therefore accessing KOSApi is disabled so now is it possible
 to register new student without problems.

hermamat committed:
Add conditional menu link for signing in/out

hermamat committed:
Request login rightaway at Homepage

hermamat committed:
Hide 'Uzivatel' link from menu when user is not logged

hermamat committed:
Fix some typos

hermamat committed:
Fix password reset link

hermamat committed:
Change trending tags interval to 2 years (for testing purposes)

hermamat committed:
Add Post form to Search Page, through which user can save new post to DB

For now all other values (except $message) are hardcoded in PostManager.
 DB model needs to be changed, because post's primary key is ID which is same as FB post, so no-FB post cannot be saved (without bugs)

hermamat committed:
Update all packages through 'composer update' and fix all dependencies

Ouch, that was the most stupid idea I've ever head - run 'composer update' on more than a year old project
and update every package at once - including base Nette framework or Nextras ORM, where was SO MANY changes!
But I figured out everything after all. Even that stupid bug with non-connecting comments was resolved...

There is a list of added/removed/updated packages:

Removed packages:
- guzzle/common (v3.9.2)
- guzzle/stream (v3.9.2)
- guzzle/parser (v3.9.2)

Added packages:
+ nextras/dbal (v1.0.1)

Updated packages:
tracy/tracy (v2.2.3) -> (v2.3.5)
latte/latte (v2.2.3) -> (v2.3.6)
nette/utils (v2.2.2) -> (v2.3.6)
nette/tokenizer (v2.2.0) -> (v2.2.1)
nette/security (v2.2.0) -> (v2.3.1)
nette/safe-stream (v2.2.0) -> (v2.3.1)
nette/finder (v2.2.0) -> (v2.3.1)
nette/caching (v2.2.1) -> (v2.3.3)
nette/robot-loader (v2.2.1) -> (v2.3.1)
nette/reflection (v2.2.0) -> (v2.3.1)
nette/php-generator (v2.2.1) -> (v2.3.4)
nette/neon (v2.2.2) -> (v2.3.3)
nette/mail (v2.2.2) -> (v2.3.3)
nette/http (v2.2.1) -> (v2.3.3)
nette/component-model (v2.2.0) -> (v2.2.4)
nette/forms (v2.2.2) -> (v2.3.5)
nette/di (v2.2.1) -> (v2.3.6)
nette/deprecated (v2.2.0) -> (v2.3.1)
nette/database (dev-master 28dff36) -> (v2.3.7)
nette/bootstrap (v2.2.2) -> (v2.3.3)
nette/application (v2.2.2) -> (v2.3.7)
nette/nette (dev-master e23de7a) -> (v2.3.7)
nextras/orm (dev-master e604182) -> (v1.1.4)
nextras/migrations (v3.0.0-rc2) -> (v3.0.2)
nextras/link-factory (v1.0.0) -> (v1.1.0)
nextras/secured-links (v1.2.0) -> (v1.3.1)
kdyby/events (v2.2.0) -> (v2.4.0)
dibi/dibi (v2.2.2) -> (v2.3.4)
symfony/event-dispatcher (v2.5.5) -> (v2.7.6)
guzzle/http (v3.9.2) -> (v3.9.3)
pimple/pimple (v2.1.1) -> (v3.0.2)
monolog/monolog (1.11.0) -> (1.17.2)
elasticsearch/elasticsearch (v1.2.2) -> (v1.4.1)
symfony/console (v2.5.5) -> (v2.7.6)
kdyby/console (v2.3.0) -> (v2.5.0)
facebook/php-sdk-v4 (4.0.11) -> (4.0.23)
kdyby/curl-ca-bundle (v1.0.5) -> (v1.0.9)
nette/tester (v1.3.0) -> (v1.6.1)
hamcrest/hamcrest-php (v1.2.2)
mockery/mockery (0.9.2) -> (0.9.4)
nextras/static-router (dev-master 88a80bf) -> (v1.0.1)

hermamat committed:
Change utf8_czech_ci collation to utf8mb4_czech_ci collation

MySQL utf8 collation doesn't support full range of unicode characters, so when there was special unicode
character in message, FB crawler just shut down with exception 'incorrect string value'.
With utf8mb4 collation it is ok.

hermamat committed:
Allow showing app inside an out-of-domain iframe (set http frames to yes)

hermamat committed:
Fix utf8 problem with changing Dbal charset to utf8mb4 too

hermamat committed:
Replace hardcoded graphVersion variable with proper from config

 In Facebook crawler there was hardcoded version for FB graphAPI (v2.3).
 I have connected proper variable in config file and sent it to constructor as parameter.

hermamat committed:
Refactor code

hermamat committed:
Refactor and comment BasePresenter

hermamat committed:
Move findRecent() method for news from mapper to repository

It is better to have the query DB independent, if it is possible, so now findRecent() is rewritten with
Nextras ORM methods and putted to Repository.

hermamat committed:
Update used FB API from 2.3 to 2.5

In old API there was problem with some posts which type was resolved as post even
if it has photo or link - this lead to not showing that photo or link.
Updating to newer API solved this problem.

hermamat committed:
Start indexing replies to comments

 FB introduced "new" feature - reply to comment, which was not crawled by fiťák,
 so I've added new foreach which indexes replies to comments (if there are any).

hermamat committed:
Fix wrongly deleted "createdTemplate" in BasePresenter

Because of bad knowledge of not-mine code I've removed createTemplate method in
 BasePresenter and replaced it with beforeRender method.
 It started to throw exception when parsing @layout.latte because it doesn't know "assert" parameter.
 It was because createTemplate function registers custom filters and macros from TemplateFilters.php and
 TemplateMacros.php files.

hermamat committed:
Show replies to comments in template

Retrieve replies to comments from db, send them to StreamControl template
and show them inside inner div

hermamat committed:
Fix permalinks in posts/comments/replies

 Links which lead to FB profiles, groups, posts or comments were broken due to old API.

hermamat committed:
Add console parameters for debugging purposes to crawler:facebook

When debugging, there is no need to crawl all the posts in group, so I added
two console parameters:
--flushLimit - tells after how many persists data should be flushed to DB (default 50)
--exit - if set, it exits right after flushing data to DB

hermamat committed:
Fix crawling of comment attachments so photos inside comments are shown

When post has attachment, its type is changed to photo/link/... and there is corresponding
fields added, but comments don't have a type field and when there is a attachment to comment/reply,
new field attachment is added, so it needs to be handled differently.

hermamat committed:
Add other types of attachment

There are some other types of attachment, which doesn't correspond with types of statuses
and they are not documented, because on FB Graph API documentation they state that
"attachment" field is deprecated and not used anymore, but it isn't and is still used.
There is no list of available attachment types, so some types could be unrecognised.
In that case they are labeled as "status" which means there is no attachment showed.

hermamat committed:
Revert changes in migrations

Because of my lack of knowledge about migrations (and not thinking about it) I've changed
old migration scripts, which broke whole concept of migrations.

hermamat committed:
Fix key-value migration - set default character set to utf8

There was no default character set set for key-value migration and it relied on
default database character set. When it wasn't utf8, but utf8mb4, migration
throwed exception "Specified key was too long; max key length is 767 bytes".

hermamat committed:
Change data charset from utf8 to utf8mb4

Because utf8 in MySQL cannot save some utf-8 characters, when there was some special
character in FB message or whatever, FB crawler throwed an exception.
I've set default charset for all tables to utf8mb4 and collation to utf8mb4_czech_ci and then
I've removed  all charset settings from all columns so they inherit it from table defaults.

 InnoDB restrict max 767 bytes for primary keys, so I've needed to lower character limit
 in tables where varchar was primary key.
  (utf8 has 3 B -> 767 / 3 ~= 255 -> max varchar(255),
  utf8mb4 has 4 B -> 767 / 4 ~= 191 -> max varchar(191)).

hermamat committed:
Add hermamat user to dummy-data in migrations

For testing purposes, so I don't need to register every time I drop the database.

hermamat committed:
Fix default value for flushLimit (FB crawler)

hermamat committed:
Change primary key in data table to internal auto_increment id

In order to save non-fb posts and comments, there needs to be set internal db id
 which doesn't rely fully on FB post/comment id.
 Therefore primary key in data needed to be changed to basic auto_increment id and the old id was
 set to fb_id which is unique, but can be NULL (posts which are not synchronised to FB groups).

Because foreign keys were changed (parent_id in data and post_id in data_tags) to other type,
there is no way (or I don't know about any) how to make it work with existing data only with SQL.
Therefore all data in `data` and `data_tags` table needs to be removed before this migration.
Only way I can think of migrating the data is write small external script which would reindex all rows in
 data and data_tags tables from fb_id to new primary id.

 For testing purposes there is migration `2015-12-09-001-truncate-all-data.sql` (so migrations can go fluently)
 which is horrible idea for deployment (just remove all data silently...) so .gitignore is added.

 //TODO find better solution how to migrate this...

hermamat committed:
Change primary key in data table to internal auto_increment id

In order to save non-fb posts and comments, there needs to be set internal db id
 which doesn't rely fully on FB post/comment id.
 Therefore primary key in data needed to be changed to basic auto_increment id and the old id was
 set to fb_id which is unique, but can be NULL (posts which are not synchronised to FB groups).

Because foreign keys were changed (parent_id in data and post_id in data_tags) to other type,
there is no way (or I don't know about any) how to make it work with existing data only with SQL.
Therefore all data in `data` and `data_tags` table needs to be removed before this migration.
Only way I can think of migrating the data is write small external script which would reindex all rows in
 data and data_tags tables from fb_id to new primary id.

 For testing purposes there is migration `2015-12-09-001-truncate-all-data.sql` (so migrations can go fluently)
 which is horrible idea for deployment (just remove all data silently...) so .gitignore is added.

 //TODO find better solution how to migrate this...

hermamat committed:
Merge remote-tracking branch 'origin/dev' into dev

Conflicts:
	migrations/structures/2015-12-09-002-change-data-primary-key.sql

hermamat committed:
Fix primary key migration (add auto increment)

hermamat committed:
Add gitignore so dev migrations are not commited

hermamat committed:
Add newline to gitignore

hermamat committed:
Add backslashes to .gitignore for dev migrations

hermamat committed:
Add FB token migration to gitignore

hermamat committed:
Change FB crawler according to database changes (add fb_id)

Change FB crawler so it works with internal int ID but can search and compares new
posts/comments according to fb_id.

hermamat committed:
Remove id in data_tags table and change primary keys

Id in data_tags is absolutely unecessary - it is table only for m:n relation and it is not used, so
I deleted it and set pair data_id-tags_id as primary key.

 BTW: In previous commit I've fixed adding post-tags relation to ORM - I've replaced add
 with set - add always add tag even if it already exists - then there was problems with duplicates.
 Set always set proper tags and call add if they don't exist or delete if they shouldn't exist anymore.

hermamat committed:
Remove forgotten debug line (persistAndFlush every time instead only persist)

hermamat committed:
Refactor printing to console for FB crawler

Printing to console was really messy after debugging, so I moved everything into indexGroup,
made two helper functions and set two levels of verbosity:
normal - prints only one letter - for normal use and checking, that crawler is still crawling
verbose - for debugging purposes - prints post ids, updated time and, part of message

hermamat committed:
Fix FB links in posts/comments/replies

Because of transformation from primary FB ids to internal ids, links broke, so they needed to be
rewritten so they links to fb_id, not id.

hermamat committed:
Set target="_blank" for links which goes to FB

I think it's less confusing for user when links which goes out of Fiťák are opened in new tab.

hermamat committed:
Merge branch 'unfinished' into dev

Conflicts:
	app/config/common.neon

hermamat committed:
Add post form in Search presenter for writing posts (unfinished)

hermamat committed:
Separate all author info from data to users table

I moved info about authors from data table (from_id and from_name) to users table,
because most of the authors would be also users. It is better for three reasons:
1) Author data is separated from post - makes logically better sense
2) Author data won't be duplicated
3) Author will be connected to exact user - better for handling user info etc.

I've created new column "registered" in user's table.
When it is 1, user is registered and can have access
to fitak and has his own user profile. When it is 0, it's only author which is crawled from FB,
but when this person create account and connect it with FB, it will match his author profile
(not implemented yet).

I also changed corresponding things in ORM, FB API, FB crawler, stream template, elastic
so everything works now.

hermamat committed:
Merge branch 'postForm' into dev

Conflicts:
	app/models/orm/Posts/Post.php

hermamat committed:
Set group_id optional for non-FB posts

hermamat committed:
Add post form for posting messages

Now logged user can post messages from search page.

hermamat committed:
Add logo for FB permission submission

When asking for some permission, FB needs some info about the App to be filled
and one of them is App logo, which should have 1024x1024 px.

hermamat committed:
Refactor stream template and add hiding replies

Refactor topics/posts, comments and replies so all has united "entry" template and the differences
are made by conditional "type" parameter. It is better because all levels
(posts-comments-replies) have share most of the fields so it prevents code duplication.

hermamat committed:
Add new dev migration (only my own test group)

 For testing purposes I've changed default group to my own testing group,
 so I've added this migration to gitignore.

hermamat committed:
[dev] Switch compiled css a js for non-compiled css and jss

 When working on css and js it's annoying to compile it everytime, so I've changed
 compiled css and js for hardcoded non-compiled files.
 It's really not the best way to commit it, but I don't know how to make it better for now
 (or better I don't have to spend time to figure it now..)

hermamat committed:
Change search GUI to more compact and FB-like feel

Old GUI was taking so much space (especially when comments and replies were opened)
so I've changed to something more compact.
For now it is more for testing purposes, not really the final version.

hermamat committed:
Add comment functionality (not FB)

 Now user have the possibility to comment posts and reply to comments.
 It is not checking if it's FB post or not and it doesn't post back to FB so now
 it make posts from FB desynchronized.

tajo commented:
Cau, promin, ze jsem si nevsiml tohodle PR drive. O dev verzi nemam vubec prehled a autori @JanTvrdik a @Mikulas uz s tim asi taky nic delat nebudou. Chces se projektu ujmout? Jak daleko je to od releasu? Diky.

hermamat committed:
Fix FB login button

I've changed FacebookId to FbId in ORM model before so it stopped working.
 Now I renamed even the calling part.
 Secondly I've fixed parameters in FB Javascript API call (when pressing FB button)

hermamat committed:
Add profile picture to FB login JS API retrieve

hermamat committed:
Connect user profile with FB account when registering

There needed to be possibility to connect FB account with internal user account in Fitak
for ability to make posts which would be send back to FB group and possibility to
comment/reply to comments on posts which are crawled from FB.

FB login doesn't supply internal user account (so it's not the another way to log into Fitak),
but it retrieves FB id, user access token, FB name and profile picture and add that informations
into register form.

When there is already some non-registered user in DB (it means that somebody written into FB group
which was crawled by Fitak afterwards, but he doesn't have Fitak account yet) and if he connect his account
with FB, it will match corresponding user with the same ID and just fill more informations to this profile.

There is some other bug fixes (as catching exceptions when storing duplicate entry to DB or when sending
 e-mail doesn't work properly).

hermamat committed:
Delete signUpConfirm.latte

This template is not used because after confirming e-mail address user is
redirected to Homepage right away.

hermamat committed:
Add anonymous user image

Default anonymous image for users which doesn't have their own profile pictures.

hermamat committed:
Move Comment/PostManager models into component

Because I've totally misunderstood how components work, I've made models with methods,
 which should be inside components itself.
 So now I putted it back where it belongs.

hermamat committed:
Clean up user  dependency passing into CommentForm

Instead of passing not used User from Nette\Security pass only found
Fitak\User from ORM.

hermamat committed:
Unfinished connection to Kdyby

hermamat committed:
Fix fatal error when access token has no expiration

If requested access token when authentificating and using Fitak crawler had
no expiration, auth and crawler scripts crashed on fatal error.

hermamat committed:
Fix fatal error when access token has no expiration

If requested access token when authentificating and using Fitak crawler had
no expiration, auth and crawler scripts crashed on fatal error.

hermamat committed:
Retrieve extended FbAccessToken right after sign up

Because JS FB SDK login which is called by button in sign up form returns only
 short-term access token, before saving the token into database, call FB api from server side
 and retrieve long-term access token which is saved into DB.

hermamat committed:
Merge branch 'commentToFBKdyby' into dev

hermamat committed:
Comments and replies are send back to FB, when it's a FB post

Finally! Comments and replies are sending back.
BTW there is really not much catching exceptions..TODO

hermamat committed:
Request user_groups and publish_actions permissions

When connecting user account with FB, request two special permissions -
user_groups and public_actions, so he can post/comment in all groups
he is a member of.

hermamat committed:
[+] Post posts to FB group

I've add functionality for sending posts back to Facebook group.
In UI user check he wants to post to FB group and from list he choose in which group it should
be posted.

hermamat committed:
Add missing files to vendor

When I was updating vendors with composer update, I've forgot to add new files.
So I'm adding them now...

hermamat committed:
Delete whitespace

hermamat committed:
Fix elastic search updater which wasn't updating posts properly

When something was updated in database, elasticsearch reindexed the newly updated post,
but it missed adding some data, so afterwards these posts weren't seen.

hermamat committed:
Fix commenting so comment update the time of parent post

When somebody comment/reply to comment, it should update the update time
in parent comment/post, so it could reorder the active posts.

hermamat committed:
Update gitignore so it ignores elasticdump

I've added elasticdump for debugging purposes and it doesn't need to be in CVS.

hermamat committed:
Add votes table into database migrations

Because I want to implement voting mechanism, I've added new "votes" table which will store all
votes and their relations between entries and users so application exactly know who up/down vote
which entry and so it prevents the same user to make more votes to same entry, or to change/remove
vote anytime later.

hermamat committed:
Add upvote functionality

Add ability to upvote entry. Now user can upvote some posts/comments.
Each user has only one vote per entry.
This functionality is prerequisite for Q&A type of posts (it will sort according to votes), but it can
be treated as likes in normal FB-style posts/comments.

hermamat committed:
Add also downvote functionality

Add functionality to downvote entry.
It's better to have bidirectional voting to distinguish great and poor user entries better.

hermamat committed:
Count and show number of votes

Count positive and negative votes and show them next to entry.

hermamat committed:
Rewrite topic retrieving to use Nextras ORM

There was some leftovers for Dibi connection, so now I retrieve posts through Post model
 which is under Nextras ORM instead of Data model which used Dibi.

 Everything was much more simplified, because Nextras ORM have a relations right from
 default (if set up correctly) so no more retrieving every connection to other models.

 //TODO I've broken retrieving $marked property, so it's need to be rewritten later (sorry, no time now..)

hermamat committed:
Spaces to tabs

hermamat committed:
Fix wrong name in migrations file

I've swapped month and day in migrations filename.

hermamat committed:
Add Q&A type and sort its answers by number of votes

When creating post, user can choose whether his post is Q&A type or not.
If it is, then its comments (i.e. Q&A answers) aren't sorted by created type but
number of votes.

This is very unclean solution, it needs to be rewritten.

hermamat committed:
Sorting by votes is moved to ORM Post entity

Sorting by votes function is put where it belongs - into ORM entity.

For now it still isn't really clean, because ICollection can't sort by virtual property,
but only by DB columns. Therefore there needs to be column which store counted int
of votes. It is needed to get everything totally synchronised (voting up, down,
removing votes) so there is not this functionality for now. Maybe later...

hermamat committed:
Change local example NEON file

hermamat committed:
Move SearchPresenter to Homepage

hermamat committed:
Add Foundation framework

This is fully installed Foundation framework with all components and requirements

bower_compontents and node_modules aren't versioned
but it is compilled with Foundation CLI interface
It would be great to strip it down and integrate it with project GULP commands, but for now I don't have time for it
and I don't fully understand how Gulp, Bower and all of that utilities workflow works...

hermamat committed:
Move old templates to another folder, start making new template by copying working static HTML

hermamat committed:
Separate layout from content

hermamat committed:
Working hiding and showing answers and comments

hermamat committed:
Some small tweaks

hermamat committed:
Comments and replies seems working

hermamat committed:
Working comments and replies

hermamat committed:
Working all about posts

hermamat committed:
Working sending messages, even QA

hermamat committed:
Add delete post menu

hermamat committed:
Fix search form

hermamat committed:
Don't show deleted posts..but now elastic fulltext search is not searching in comments and replies

hermamat committed:
Delete comments - only for admin

hermamat committed:
Fix children only when not deleted

hermamat committed:
Commenting on post working

hermamat committed:
Replying to comment working - at least the main thing

hermamat committed:
Fix answers and replies and everything..

hermamat committed:
Add tag cloud and make font smaller

hermamat committed:
Add savedSearches

hermamat committed:
Add savedSearches handle - not working

hermamat committed:
Move Search component creation to Base presenter

hermamat committed:
Fit static templates into containers

hermamat committed:
Add Help and Admin pages

hermamat committed:
Saving searches working

hermamat committed:
Add savedSearches control (viewing saved searches)

hermamat committed:
Working even deleting saved searches

hermamat committed:
Add possibility to remove votes

hermamat committed:
Show current user votes (in main posts)

hermamat committed:
Working voting even for all children

hermamat committed:
Admin page with adding another (even non-cvut) user, making somebody admin

hermamat committed:
Admin page with adding and removing FB groups

hermamat committed:
Update Nette Tester

hermamat committed:
Add test files to ignore

hermamat committed:
Add possibility to disable group for crawling, add DB snapshot with proper structure and test data

hermamat committed:
Add warning banner to Sign In about only testing version
